app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -234,4 +234,23 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
             updateOverlayForCurrentState()
         }
     }
+
+    override fun onNonTextInputUrlEntered(urlStr: String) = with (navUrlInput) {
+        if (urlStr.isNotEmpty()) {
+            val cachedAutocompleteResult =
+                    lastAutocompleteResult // setText clears the reference so we cache it here.
+            setText(cachedAutocompleteResult.text)
+            onNavigationEvent?.invoke(NavigationEvent.LOAD, urlStr, cachedAutocompleteResult)
```

Comment:
Looking through the NavigationEvent LOAD code here, it eventually calls into `MainActivity.onTextInputUrlEntered` which will send the wrong telemetry. (This code is kind of confusing, so I also filed #800 )